### PR TITLE
chore(deps): update to eslint v9

### PR DESCRIPTION
**Pull request by dominikg**
used the migration tool but had to do a few changes after

The `node: false` in globals is replaced by a mapped 'off' spread, but that came after the browser spread, disabling shared globals in the process.

todo: remove the FlatCompat helper

### CODE REVIEW

**Review by bluwy — APPROVED** [review]
I don't understand every config, but seems ok 😄

**Comment by dominikg** [comment]
@benmccann  does this look ok? esp. the integration of the svelte config? I am a bit surprised that it doesn't setup the ts parser, or does that work ootb now?

**Comment by dominikg** [comment]
oh, and it having browser and node globals enabled for everything by default is a bit lax too o_O

File: eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+				'error',
+				{
+					allowModules: ['types', 'estree', 'testUtils', '@sveltejs/vite-plugin-svelte', 'svelte']
+				}
+			],
+
+			'n/no-missing-require': [
+				'error',
+				{
+					allowModules: ['pnpapi']
```
Comment (benmccann) [inline eslint.config.js:58]: is this still needed? I didn't see `pnpapi` in the code

File: eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+			],
+
+			'n/no-extraneous-require': [
+				'error',
+				{
+					allowModules: ['vite']
+				}
+			],
+
+			'n/no-deprecated-api': 'off',
```
Comment (benmccann) [inline eslint.config.js:76]: are we using many deprecated APIs? maybe we could just ignore them where we are or whitelist them here if they can't be fixed?

File: eslint.config.js [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+	{
+		files: ['packages/playground/kit-demo-app/src/**'],
+
+		rules: {
+			'n/no-missing-import': 'off',
+			'n/no-unsupported-features/node-builtins': 'off'
+		}
+	},
+	{
+		files: ['**/*.d.ts'],
```
Comment (benmccann) [inline eslint.config.js:239]: this was already present above

**Comment by benmccann** [comment]
> oh, and it having browser and node globals enabled for everything by default is a bit lax too o_O

I don't remember how many places we'd need to enable them if they're not enabled globally or how hard that would be too do. We could investigate. As long as we don't make it so annoying that Rich rips out the whole eslint setup out of frustration :laughing:

File: eslint.config.js (hunk 1 above)
Comment (dominikg) [reply eslint.config.js:58]: honestly don't remember why we had this. @bluwy  was it sth that moved with vitefu?

Comment (bluwy) [reply eslint.config.js:58]: Seems to exist for a very long time when the plugin is copied from the Vite repo. Likely safe to remove this now.

File: eslint.config.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
+import markdown from 'eslint-plugin-markdown';
+import globals from 'globals';
+import n from 'eslint-plugin-n';
+import svelte_config from '@sveltejs/eslint-config';
+import svelteParser from 'svelte-eslint-parser';
```
Comment (benmccann) [inline eslint.config.js:5]: this is camelCase but line above is snake_case

**Pull request merged**